.gitignore

```diff
@@ -44,6 +44,8 @@ docker-images/operator/tmp/**
 
 # Connect plugin (tests)
 systemtest/*.tar.gz
+systemtest/bin/
+cluster-operator/bin/
```

Comment:
Why is this needed? I do not have anything in these directories that would need to be ignored.

Reply:
the `make` tasks generated a lot of files inside these folders which I do not want to commit!

Reply:
That is strange because it does not create these in other environments.

Reply:
```
systemtest/bin
├── Makefile
├── pom.xml
├── scripts
│   ├── results_info.sh
│   └── run_tests.sh
├── src
│   ├── main
│   │   ├── java
│   │   │   └── io
│   │   │       └── strimzi
│   │   │           └── systemtest
│   │   │               ├── annotations
│   │   │               │   ├── FIPSNotSupported.class
│   │   │               │   ├── FIPSNotSupportedCondition.class
│   │   │               │   ├── IPv6NotSupported.class
│   │   │               │   ├── IPv6NotSupportedCondition.class
│   │   │               │   ├── IsolatedTest.class
│   │   │               │   ├── KindIPv6NotSupported.class
│   │   │               │   ├── KindIPv6NotSupportedCondition.class
│   │   │               │   ├── KindNotSupported.class
│   │   │               │   ├── KindNotSupportedCondition.class
│   │   │               │   ├── KRaftNotSupported.class
│   │   │               │   ├── KRaftNotSupportedCondition.class
│   │   │               │   ├── MicroShiftNotSupported.class
│   │   │               │   ├── MicroShiftNotSupportedCondition.class
│   │   │               │   ├── MixedRoleNotSupported.class
│   │   │               │   ├── MixedRoleNotSupportedCondition.class
│   │   │               │   ├── MultiNodeClusterOnly.class
│   │   │               │   ├── MultiNodeClusterOnlyCondition.class
│   │   │               │   ├── NodePoolsNotSupported.class
│   │   │               │   ├── NodePoolsNotSupportedCondition.class
│   │   │               │   ├── NodePoolsOnly.class
│   │   │               │   ├── NodePoolsOnlyCondition.class
│   │   │               │   ├── OpenShiftOnly.class
│   │   │               │   ├── OpenShiftOnlyCondition.class
│   │   │               │   ├── ParallelNamespaceTest.class
│   │   │               │   ├── ParallelTest.class
│   │   │               │   ├── RequiredMinKubeApiVersion.class
│   │   │               │   ├── RequiredMinKubeApiVersionCondition.class
│   │   │               │   ├── RequiredMinKubeOrOcpBasedKubeVersion.class
│   │   │               │   ├── RequiredMinKubeOrOpenshiftVersionCondition.class
│   │   │               │   └── SkipDefaultNetworkPolicyCreation.class
│   │   │               ├── cli
│   │   │               │   └── KafkaCmdClient.class
│   │   │               ├── enums
│   │   │               │   ├── ClusterOperatorInstallType.class
│   │   │               │   ├── ClusterOperatorRBACType.class
│   │   │               │   ├── ConditionStatus.class
│   │   │               │   ├── CustomResourceStatus.class
│   │   │               │   ├── DefaultNetworkPolicy.class
│   │   │               │   ├── DeploymentTypes.class
│   │   │               │   ├── NodePoolsRoleMode.class
│   │   │               │   ├── OlmInstallationStrategy.class
│   │   │               │   ├── PodSecurityProfile.class
│   │   │               │   └── UserAuthType.class
│   │   │               ├── Environment.class
│   │   │               ├── exceptions
│   │   │               │   └── KubernetesClusterUnstableException.class
│   │   │               ├── interfaces
│   │   │               │   └── IndicativeSentences.class
│   │   │               ├── k8s
│   │   │               │   └── Events.class
│   │   │               ├── kafkaclients
│   │   │               │   ├── AbstractKafkaClient$Builder.class
│   │   │               │   ├── AbstractKafkaClient.class
│   │   │               │   ├── clientproperties
│   │   │               │   │   ├── AbstractKafkaClientProperties$KafkaClientPropertiesBuilder.class
│   │   │               │   │   ├── AbstractKafkaClientProperties.class
│   │   │               │   │   ├── ConsumerProperties$ConsumerPropertiesBuilder.class
│   │   │               │   │   ├── ConsumerProperties.class
│   │   │               │   │   ├── ProducerProperties$ProducerPropertiesBuilder.class
│   │   │               │   │   └── ProducerProperties.class
│   │   │               │   ├── externalClients
│   │   │               │   │   ├── ExternalKafkaClient$Builder.class
│   │   │               │   │   └── ExternalKafkaClient.class
│   │   │               │   └── internalClients
│   │   │               │       ├── admin
│   │   │               │       │   ├── AdminClient$AdminTopicCommand.class
│   │   │               │       │   ├── AdminClient.class
│   │   │               │       │   ├── KafkaAdminException.class
│   │   │               │       │   └── KafkaTopicDescription.class
│   │   │               │       ├── BaseClients.class
│   │   │               │       ├── BridgeClients.class
│   │   │               │       ├── BridgeTracingClients.class
│   │   │               │       ├── KafkaClients.class
│   │   │               │       ├── KafkaOauthClients.class
│   │   │               │       └── KafkaTracingClients.class
│   │   │               ├── keycloak
│   │   │               │   └── KeycloakInstance.class
│   │   │               ├── listeners
│   │   │               │   └── ExecutionListener.class
│   │   │               ├── logs
│   │   │               │   ├── LogCollector.class
│   │   │               │   └── TestExecutionWatcher.class
│   │   │               ├── matchers
│   │   │               │   ├── HasAllOfReasons.class
│   │   │               │   ├── HasAnyOfReasons.class
│   │   │               │   ├── HasNoneOfReasons.class
│   │   │               │   ├── LogHasNoUnexpectedErrors$LogIgnoreList.class
│   │   │               │   ├── LogHasNoUnexpectedErrors.class
│   │   │               │   └── Matchers.class
│   │   │               ├── metrics
│   │   │               │   ├── MetricsCollector$Builder.class
│   │   │               │   └── MetricsCollector.class
│   │   │               ├── parallel
│   │   │               │   ├── SuiteThreadController.class
│   │   │               │   └── TestSuiteNamespaceManager.class
│   │   │               ├── performance
│   │   │               │   ├── gather
│   │   │               │   │   ├── collectors
│   │   │               │   │   │   ├── BaseMetricsCollector.class
│   │   │               │   │   │   ├── TopicOperatorMetricsCollector$Builder.class
│   │   │               │   │   │   ├── TopicOperatorMetricsCollector.class
│   │   │               │   │   │   ├── UserOperatorMetricsCollector$Builder.class
│   │   │               │   │   │   └── UserOperatorMetricsCollector.class
│   │   │               │   │   └── schedulers
│   │   │               │   │       ├── BaseMetricsCollectionScheduler.class
│   │   │               │   │       ├── TopicOperatorMetricsCollectionScheduler.class
│   │   │               │   │       └── UserOperatorMetricsCollectionScheduler.class
│   │   │               │   ├── OperationTimer.class
│   │   │               │   ├── PerformanceConstants.class
│   │   │               │   ├── report
│   │   │               │   │   ├── BasePerformanceReporter.class
│   │   │               │   │   ├── parser
│   │   │               │   │   │   ├── BasePerformanceMetricsParser.class
│   │   │               │   │   │   ├── ExperimentMetrics.class
│   │   │               │   │   │   ├── ParserFactory.class
│   │   │               │   │   │   ├── ParserType.class
│   │   │               │   │   │   ├── TopicOperatorMetricsParser.class
│   │   │               │   │   │   └── UserOperatorMetricsParser.class
│   │   │               │   │   ├── TopicOperatorPerformanceReporter.class
│   │   │               │   │   └── UserOperatorPerformanceReporter.class
│   │   │               │   └── utils
│   │   │               │       └── UserOperatorPerformanceUtils.class
│   │   │               ├── resources
│   │   │               │   ├── ComponentType.class
│   │   │               │   ├── crd
│   │   │               │   │   ├── KafkaBridgeResource.class
│   │   │               │   │   ├── KafkaClientsResource.class
│   │   │               │   │   ├── KafkaConnectorResource.class
│   │   │               │   │   ├── KafkaConnectResource.class
│   │   │               │   │   ├── KafkaMirrorMaker2Resource.class
│   │   │               │   │   ├── KafkaMirrorMakerResource.class
│   │   │               │   │   ├── KafkaNodePoolResource.class
│   │   │               │   │   ├── KafkaRebalanceResource.class
│   │   │               │   │   ├── KafkaResource.class
│   │   │               │   │   ├── KafkaTopicResource.class
│   │   │               │   │   ├── KafkaUserResource.class
│   │   │               │   │   └── StrimziPodSetResource.class
│   │   │               │   ├── draincleaner
│   │   │               │   │   ├── DrainCleanerResource.class
│   │   │               │   │   └── SetupDrainCleaner.class
│   │   │               │   ├── jaeger
│   │   │               │   │   └── SetupJaeger.class
│   │   │               │   ├── keycloak
│   │   │               │   │   └── SetupKeycloak.class
│   │   │               │   ├── kubernetes
│   │   │               │   │   ├── ClusterOperatorCustomResourceDefinition.class
│   │   │               │   │   ├── ClusterRoleBindingResource.class
│   │   │               │   │   ├── ClusterRoleResource.class
│   │   │               │   │   ├── ConfigMapResource.class
│   │   │               │   │   ├── DeploymentResource.class
│   │   │               │   │   ├── JobResource.class
│   │   │               │   │   ├── LeaseResource.class
│   │   │               │   │   ├── NetworkPolicyResource.class
│   │   │               │   │   ├── RoleBindingResource.class
│   │   │               │   │   ├── RoleResource.class
│   │   │               │   │   ├── SecretResource.class
│   │   │               │   │   ├── ServiceAccountResource.class
│   │   │               │   │   ├── ServiceResource.class
│   │   │               │   │   └── ValidatingWebhookConfigurationResource.class
│   │   │               │   ├── NamespaceManager.class
│   │   │               │   ├── NodePoolsConverter.class
│   │   │               │   ├── openshift
│   │   │               │   │   ├── OperatorGroupResource.class
│   │   │               │   │   └── SubscriptionResource.class
│   │   │               │   ├── operator
│   │   │               │   │   ├── BundleResource$BundleResourceBuilder.class
│   │   │               │   │   ├── BundleResource.class
│   │   │               │   │   ├── configuration
│   │   │               │   │   │   └── OlmConfiguration.class
│   │   │               │   │   ├── SetupClusterOperator$SetupClusterOperatorBuilder.class
│   │   │               │   │   ├── SetupClusterOperator.class
│   │   │               │   │   └── specific
│   │   │               │   │       ├── HelmResource.class
│   │   │               │   │       ├── OlmResource.class
│   │   │               │   │       └── SpecificResourceType.class
│   │   │               │   ├── ResourceCondition.class
│   │   │               │   ├── ResourceItem.class
│   │   │               │   ├── ResourceManager.class
│   │   │               │   ├── ResourceOperation.class
│   │   │               │   ├── ResourceType.class
│   │   │               │   └── ThrowableRunner.class
│   │   │               ├── security
│   │   │               │   ├── CertAndKeyFiles.class
│   │   │               │   ├── OpenSsl$OpenSslCommand.class
│   │   │               │   ├── OpenSsl.class
│   │   │               │   ├── SystemTestCertAndKeyBuilder.class
│   │   │               │   ├── SystemTestCertAndKey.class
│   │   │               │   ├── SystemTestCertHolder.class
│   │   │               │   └── SystemTestCertManager.class
│   │   │               ├── storage
│   │   │               │   └── TestStorage.class
│   │   │               ├── templates
│   │   │               │   ├── crd
│   │   │               │   │   ├── KafkaBridgeTemplates.class
│   │   │               │   │   ├── KafkaConnectorTemplates.class
│   │   │               │   │   ├── KafkaConnectTemplates.class
│   │   │               │   │   ├── KafkaMirrorMaker2Templates.class
│   │   │               │   │   ├── KafkaMirrorMakerTemplates.class
│   │   │               │   │   ├── KafkaNodePoolTemplates.class
│   │   │               │   │   ├── KafkaRebalanceTemplates.class
│   │   │               │   │   ├── KafkaTemplates.class
│   │   │               │   │   ├── KafkaTopicTemplates.class
│   │   │               │   │   └── KafkaUserTemplates.class
│   │   │               │   ├── kubernetes
│   │   │               │   │   ├── ClusterRoleBindingTemplates.class
│   │   │               │   │   ├── NetworkPolicyTemplates.class
│   │   │               │   │   └── ServiceTemplates.class
│   │   │               │   └── specific
│   │   │               │       ├── AdminClientTemplates.class
│   │   │               │       └── ScraperTemplates.class
│   │   │               ├── TestConstants.class
│   │   │               ├── tracing
│   │   │               │   └── TracingConstants.class
│   │   │               ├── upgrade
│   │   │               │   ├── BundleVersionModificationData.class
│   │   │               │   ├── CommonVersionModificationData.class
│   │   │               │   ├── OlmVersionModificationData.class
│   │   │               │   ├── UpgradeKafkaVersion.class
│   │   │               │   ├── VersionModificationDataLoader$ModificationType.class
│   │   │               │   └── VersionModificationDataLoader.class
│   │   │               └── utils
│   │   │                   ├── AdminClientUtils.class
│   │   │                   ├── ClientUtils.class
│   │   │                   ├── FileUtils.class
│   │   │                   ├── HttpUtils.class
│   │   │                   ├── kafkaUtils
│   │   │                   │   ├── KafkaBridgeUtils.class
│   │   │                   │   ├── KafkaConnectorUtils.class
│   │   │                   │   ├── KafkaConnectUtils.class
│   │   │                   │   ├── KafkaMirrorMaker2Utils.class
│   │   │                   │   ├── KafkaMirrorMakerUtils.class
│   │   │                   │   ├── KafkaNodePoolUtils.class
│   │   │                   │   ├── KafkaRebalanceUtils.class
│   │   │                   │   ├── KafkaTopicScalabilityUtils.class
│   │   │                   │   ├── KafkaTopicUtils.class
│   │   │                   │   ├── KafkaUserUtils.class
│   │   │                   │   └── KafkaUtils.class
│   │   │                   ├── kubeUtils
│   │   │                   │   ├── controllers
│   │   │                   │   │   ├── ConfigMapUtils.class
│   │   │                   │   │   ├── DeploymentUtils.class
│   │   │                   │   │   ├── JobUtils.class
│   │   │                   │   │   ├── StatefulSetUtils.class
│   │   │                   │   │   └── StrimziPodSetUtils.class
│   │   │                   │   └── objects
│   │   │                   │       ├── NodeUtils.class
│   │   │                   │       ├── PersistentVolumeClaimUtils.class
│   │   │                   │       ├── PodUtils.class
│   │   │                   │       ├── SecretUtils.class
│   │   │                   │       └── ServiceUtils.class
│   │   │                   ├── RollingUpdateUtils.class
│   │   │                   ├── specific
│   │   │                   │   ├── BridgeUtils.class
│   │   │                   │   ├── CruiseControlUtils$ApiResult.class
│   │   │                   │   ├── CruiseControlUtils$HttpMethod.class
│   │   │                   │   ├── CruiseControlUtils$Scheme.class
│   │   │                   │   ├── CruiseControlUtils.class
│   │   │                   │   ├── JmxUtils.class
│   │   │                   │   ├── KeycloakUtils.class
│   │   │                   │   ├── MetricsUtils.class
│   │   │                   │   ├── OlmUtils.class
│   │   │                   │   ├── ScraperUtils.class
│   │   │                   │   └── TracingUtils.class
│   │   │                   ├── StUtils.class
│   │   │                   ├── TestKafkaVersion.class
│   │   │                   └── VerificationUtils.class
│   │   └── resources
│   │       └── log4j2.properties
│   ├── rbac
│   │   └── role-edit-kafka.yaml
│   └── test
│       ├── java
│       │   └── io
│       │       └── strimzi
│       │           └── systemtest
│       │               ├── AbstractST.class
│       │               ├── bridge
│       │               │   ├── HttpBridgeCorsST.class
│       │               │   ├── HttpBridgeKafkaExternalListenersST.class
│       │               │   ├── HttpBridgeScramShaST.class
│       │               │   ├── HttpBridgeST.class
│       │               │   └── HttpBridgeTlsST.class
│       │               ├── connect
│       │               │   ├── ConnectBuilderST.class
│       │               │   └── ConnectST.class
│       │               ├── cruisecontrol
│       │               │   ├── CruiseControlApiST.class
│       │               │   ├── CruiseControlConfigurationST.class
│       │               │   └── CruiseControlST.class
│       │               ├── kafka
│       │               │   ├── ConfigProviderST.class
│       │               │   ├── dynamicconfiguration
│       │               │   │   ├── DynamicConfSharedST.class
│       │               │   │   └── DynamicConfST.class
│       │               │   ├── KafkaNodePoolST.class
│       │               │   ├── KafkaST.class
│       │               │   ├── KafkaVersionsST.class
│       │               │   ├── listeners
│       │               │   │   ├── ListenersST.class
│       │               │   │   └── MultipleListenersST.class
│       │               │   └── QuotasST.class
│       │               ├── log
│       │               │   ├── LoggingChangeST.class
│       │               │   └── LogSettingST.class
│       │               ├── metrics
│       │               │   ├── JmxST.class
│       │               │   └── MetricsST.class
│       │               ├── migration
│       │               │   └── MigrationST.class
│       │               ├── mirrormaker
│       │               │   ├── MirrorMaker2ST.class
│       │               │   └── MirrorMakerST.class
│       │               ├── olm
│       │               │   ├── OlmAbstractST.class
│       │               │   ├── OlmAllNamespaceST.class
│       │               │   └── OlmSingleNamespaceST.class
│       │               ├── operators
│       │               │   ├── ClusterOperatorRbacST.class
│       │               │   ├── CustomResourceStatusST.class
│       │               │   ├── FeatureGatesST.class
│       │               │   ├── LeaderElectionST.class
│       │               │   ├── MultipleClusterOperatorsST.class
│       │               │   ├── NamespaceDeletionRecoveryST.class
│       │               │   ├── NamespaceRbacScopeOperatorST.class
│       │               │   ├── PodSetST.class
│       │               │   ├── ReconciliationST.class
│       │               │   ├── RecoveryST.class
│       │               │   ├── topic
│       │               │   │   ├── ThrottlingQuotaST.class
│       │               │   │   ├── TopicReplicasChangeST.class
│       │               │   │   └── TopicST.class
│       │               │   └── user
│       │               │       └── UserST.class
│       │               ├── performance
│       │               │   ├── TopicOperatorPerformance.class
│       │               │   └── UserOperatorPerformance.class
│       │               ├── rollingupdate
│       │               │   ├── AlternativeReconcileTriggersST.class
│       │               │   ├── KafkaRollerST.class
│       │               │   └── RollingUpdateST.class
│       │               ├── security
│       │               │   ├── custom
│       │               │   │   ├── CustomAuthorizerST.class
│       │               │   │   └── CustomCaST.class
│       │               │   ├── NetworkPoliciesST.class
│       │               │   ├── oauth
│       │               │   │   ├── OauthAbstractST.class
│       │               │   │   ├── OauthAuthorizationST.class
│       │               │   │   ├── OauthPasswordGrantsST.class
│       │               │   │   ├── OauthPlainST.class
│       │               │   │   ├── OauthScopeST.class
│       │               │   │   └── OauthTlsST.class
│       │               │   ├── OpaIntegrationST.class
│       │               │   ├── PodSecurityProfilesST.class
│       │               │   └── SecurityST.class
│       │               ├── specific
│       │               │   ├── DrainCleanerST.class
│       │               │   ├── HelmChartST.class
│       │               │   ├── RackAwarenessST.class
│       │               │   └── SpecificST.class
│       │               ├── tracing
│       │               │   └── OpenTelemetryST.class
│       │               ├── upgrade
│       │               │   ├── AbstractUpgradeST.class
│       │               │   ├── kraft
│       │               │   │   ├── AbstractKRaftUpgradeST.class
│       │               │   │   ├── KRaftKafkaUpgradeDowngradeST.class
│       │               │   │   ├── KRaftStrimziDowngradeST.class
│       │               │   │   └── KRaftStrimziUpgradeST.class
│       │               │   └── regular
│       │               │       ├── KafkaUpgradeDowngradeST.class
│       │               │       ├── OlmUpgradeST.class
│       │               │       ├── StrimziDowngradeST.class
│       │               │       └── StrimziUpgradeST.class
│       │               ├── utils
│       │               │   └── KafkaVersionUtilsTest.class
│       │               └── watcher
│       │                   ├── AbstractNamespaceST.class
│       │                   ├── AllNamespaceST.class
│       │                   └── MultipleNamespaceST.class
│       └── resources
│           ├── connect-build
│           │   ├── build-connect-image.sh
│           │   └── Dockerfile
│           ├── junit-platform.properties
│           ├── META-INF
│           │   └── services
│           │       └── org.junit.platform.launcher.TestExecutionListener
│           ├── oauth2
│           │   ├── authorization_realm.json
│           │   ├── internal_realm.json
│           │   ├── keycloak-instance.yaml
│           │   ├── postgres.yaml
│           │   ├── prepare_keycloak_operator.sh
│           │   ├── scope_audience_realm.json
│           │   └── teardown_keycloak_operator.sh
│           ├── opa
│           │   └── opa.yaml
│           ├── performance
│           │   └── dashboards
│           │       └── topic-operator
│           │           └── topic-dashboard.json
│           ├── security-st-certs
│           │   ├── clients-ca.key
│           │   ├── cluster-ca.key
│           │   ├── expired-clients-ca.crt
│           │   └── expired-cluster-ca.crt
│           ├── tracing
│           │   ├── cert-manager.yaml
│           │   ├── jaeger-instance.yaml
│           │   └── jaeger-operator.yaml
│           └── upgrade
│               ├── BundleDowngrade.yaml
│               ├── BundleUpgrade.yaml
│               └── OlmUpgrade.yaml
├── target
│   ├── classes
│   │   ├── bridge.version
│   │   ├── kafka-versions.yaml
│   │   └── log4j2.properties
│   └── test-classes
│       ├── connect-build
│       │   ├── build-connect-image.sh
│       │   └── Dockerfile
│       ├── junit-platform.properties
│       ├── META-INF
│       │   └── services
│       │       └── org.junit.platform.launcher.TestExecutionListener
│       ├── oauth2
│       │   ├── authorization_realm.json
│       │   ├── internal_realm.json
│       │   ├── keycloak-instance.yaml
│       │   ├── postgres.yaml
│       │   ├── prepare_keycloak_operator.sh
│       │   ├── scope_audience_realm.json
│       │   └── teardown_keycloak_operator.sh
│       ├── opa
│       │   └── opa.yaml
│       ├── performance
│       │   └── dashboards
│       │       └── topic-operator
│       │           └── topic-dashboard.json
│       ├── security-st-certs
│       │   ├── clients-ca.key
│       │   ├── cluster-ca.key
│       │   ├── expired-clients-ca.crt
│       │   └── expired-cluster-ca.crt
│       ├── tracing
│       │   ├── cert-manager.yaml
│       │   ├── jaeger-instance.yaml
│       │   └── jaeger-operator.yaml
│       └── upgrade
│           ├── BundleDowngrade.yaml
│           ├── BundleUpgrade.yaml
│           └── OlmUpgrade.yaml
└── tmt
    ├── plans
    │   └── main.fmf
    ├── README.md
    ├── scripts
    │   └── copy-logs.sh
    └── tests
        └── strimzi
            ├── main.fmf
            └── test.sh

119 directories, 358 files
```

Is added to git when running make tasks on my machine.

Reply:
> That is strange because it does not create these in other environments.

Maybe it is a fluke on my machine. I can revert the change if you think it is unnecessary to ensure the files are ignored in .gitignore.

Reply:
That just looks like it copied everything into another subdirectory which does not make any sense. I have no idea about your environment, but this is not happening on any of my environments or in the CI. 

It could be something in your environment or some bug in the Makefiles which makes it do this in your environment? But we would need to address that and not just ignore these files.

Reply:
I will try to investigate, but in this PR I will simply revert the gitignore changes. Thanks for feedback